### PR TITLE
Migrate Dagger component builder to factory pattern

### PR DESCRIPTION
**Pull request by chibatching**
## Issue
- none

## Overview (Required)
- Migrate from Dagger component builder to factory pattern introduced from Dagger 2.22
  - Just call a single method to create components
  - It is worth to avoid forgetting to call some builder methods

## Links
- https://github.com/google/dagger/releases/tag/dagger-2.22
- https://proandroiddev.com/dagger-and-the-shiny-new-component-factory-c2234fcae6b1

## Screenshot
Before | After
:--: | :--:
 |

### CODE REVIEW

**Comment by takahirom** [comment]
👀

File: android-base/src/main/java/io/github/droidkaigi/confsched2020/di/RepositoryComponentModule.kt [hunk 1]
```diff
@@ -55,23 +55,24 @@ object RepositoryComponentModule {
...
-            .database(database)
-            .sponsorDatabase(sponsorDatabase)
-            .firestore(firestore)
-            .announcementDatabase(announcementDatabase)
-            .staffDatabase(staffDatabase)
-            .contributorDatabase(contributorDatabase)
-            .build()
+        return RepositoryComponent.factory()
+            .create(
+                context,
```
Comment (takahirom) [inline android-base/src/main/java/io/github/droidkaigi/confsched2020/di/RepositoryComponentModule.kt]: Thank you for awesome PR!
Can we use named argument for this?👀

Comment (chibatching) [reply android-base/src/main/java/io/github/droidkaigi/confsched2020/di/RepositoryComponentModule.kt]: Of course, I'll change it!

Comment (chibatching) [reply android-base/src/main/java/io/github/droidkaigi/confsched2020/di/RepositoryComponentModule.kt]: done with 5ab0afa

**Comment by takahirom** [comment]
I was able to learn from this PR!  LGTM!🎉

**Pull request merged**